map/src/menu/route/RouteMenu.js

```diff
@@ -0,0 +1,386 @@
+        '& .MuiFilledInput-root': {
+            background: '#aad3df',
+        },
+    },
+    finish: {
+        '& .MuiFilledInput-root': {
+            background: '#ebdbe8',
+        },
+    },
+});
+
+export function formatRouteInfo(props) {
+    let res = ['Route: '];
+    if (props?.overall?.distance) {
+        let dst = (props.overall.distance / 1000).toFixed(1);
+        res.push(<span key="info-dst">{dst + ' km'}</span>);
+        res.push(', ');
+    }
+    if (props?.overall?.time) {
+        let hours = props.overall.time / 3600.0;
+        let min = ((hours - Math.floor(hours)) * 60).toFixed(0);
+        if (min < 10) {
+            min = '0' + min;
+        }
+        res.push(<span key="info-time">{Math.floor(hours).toFixed(0) + ':' + min + ' min'}</span>);
+        res.push(', ');
+    }
+    res[res.length - 1] = '';
+    if (props?.overall?.routingTime) {
+        res[res.length - 1] = '.';
+        res.push(' Cost: ');
+        res.push(props.overall.routingTime.toFixed(0));
+    }
+    return <span id="se-route-info">{res}</span>;
+}
+
+function formatLatLon(pnt) {
+    if (!pnt) {
+        return '';
+    }
+    return pnt.lat.toFixed(5) + ', ' + pnt.lng.toFixed(5);
+}
+
+export default function RouteMenu() {
+    const ctx = useContext(AppContext);
+
+    const routeObject = ctx.routeObject;
+
+    const startPoint = routeObject.getOption('route.points.start');
+    const finishPoint = routeObject.getOption('route.points.finish');
+    const viaPoints = routeObject.getOption('route.points.viaPoints');
+    const avoidRoads = routeObject.getOption('route.points.avoidRoads');
+
+    const classes = useStyles();
+
+    const [open, setOpen] = useState(false);
+    const [start, setStart] = useState('');
+    const [finish, setFinish] = useState('');
+    const [openSettings, setOpenSettings] = useState(false);
+    const btnFile = useRef();
+
+    // auto switch between Navigation/Tracks menu
+    // additionally, conceal route track from the map
+    // additionally, trigger fitBounds (zoom) on route track open
+    useEffect(() => {
+        if (isRouteTrack(ctx)) {
+            setOpen(true);
+            routeObject.setOption('route.map.zoom', true);
+        } else if (isLocalTrack(ctx) || isCloudTrack(ctx)) {
+            setOpen(false);
+            ctx.routeObject.setOption('route.map.conceal', true);
+        }
+    }, [ctx.currentObjectType]);
+
+    // // auto zoom once Navigation menu open
+    // // additionally, trigger fitBounds (zoom) always on menu open
+    // // optionally, re-convert (get->put) to switch to route object type
+    // useEffect(() => {
+    //     const route = routeObject.getRoute();
+    //     if (open && route) {
+    //         if (isRouteTrack(ctx) === false) {
+    //             routeObject.putRoute({ route: routeObject.getRoute() });
+    //         } else {
+    //             routeObject.setOption('route.map.zoom', true);
+    //         }
+    //     }
+    // }, [open]);
```

Comment:
This block was used for 2 actions:

1) restore Navigation layer on the map (because Track Editor kills all "foreign" layers) - restoration was done by "reset" currentGpxFile with putRoute() -- please check is Navigation track (layer) really back when you switch back to Navigation menu after editing Local track

2) second action is Zoom-on map Navigation track when you open Navigation menu

Commenting this block breaks previously accepted task (Navigation menu changes).